staging/src/k8s.io/apiextensions-apiserver/pkg/registry/customresource/strategy.go

```diff
@@ -271,3 +278,13 @@ func (a customResourceStrategy) MatchCustomResourceDefinitionStorage(label label
 		GetAttrs: a.GetAttrs,
 	}
 }
+
+// OpenAPIv3 type/maxLength/maxItems/MaxProperties/required/wrong type field validation failures are viewed as blocking err for CEL validation
+func hasBlockingErr(errs field.ErrorList) (bool, *field.Error) {
+	for _, err := range errs {
+		if err.Type.IsCelBlockingErr() {
+			return true, field.Invalid(nil, nil, "Skipping x-kubernetes-validations rules due to OpenAPIv3 errors preventing the rules from safely executed")
```

Comment:
`x-kubernetes-validations` doesn't mean much to the CR API user... suggest something more like:

```suggestion
			return true, field.Invalid(nil, nil, "some validation rules were not checked because the object was invalid; correct the existing errors to complete validation")
```